### PR TITLE
[CORE-534] Adding backup and restore functionality for RocksDB

### PR DESCRIPTION
**Pull request by TelicentPaul**
Adding backup and restore functionality for underlying Rocks DB. 
Plus a typo fix and small unrelated test.

### CODE REVIEW

File: rdf-abac-core/src/test/java/io/telicent/jena/abac/bulk/BulkDirectory.java [hunk 1]
```diff
@@ -63,25 +69,28 @@ public abstract class BulkDirectory {
...
         level = LogCtl.getLevel(BulkDirectory.LOG);
         LogCtl.setLevel(BulkDirectory.LOG, "warn");
     }
 
-    @AfterEach public void after() {
-        if (labelsStore instanceof LabelsStoreRocksDB rocksDB)
-        {
+    @AfterEach
+    public void after() {
+        if (labelsStore instanceof LabelsStoreRocksDB rocksDB) {
```
Comment (rvesse) [inline rdf-abac-core/src/test/java/io/telicent/jena/abac/bulk/BulkDirectory.java:80]: It's sad how much this minor stylistic change makes me happy 👍

**Review by rvesse — APPROVED** [review]
LGTM, seems the RocksDB APIs are quite nice and clean here.

Thought for a future ticket - do we want to take a backup of the current labels store state prior to restoring from a backup?  Or does that feel like more of a documentation issue i.e. "if you ever think you might want to 'undo' a restore operation always take a backup prior to a restore"

File: rdf-abac-core/src/test/java/io/telicent/jena/abac/bulk/BulkDirectory.java (hunk 1 above)
Comment (TelicentPaul) [reply rdf-abac-core/src/test/java/io/telicent/jena/abac/bulk/BulkDirectory.java:80]: Agreed

**Comment by TelicentPaul** [comment]
Yeah, I went back and forth on it. Agreed, we can always do that later if we think it merited. 
Similarly, I've gone quite light with regards to locking. It should be ok but Emily's bulk testing will no doubt let us know quickly if there are issues.

On other thing that I noticed - and almost wrote tests to illustrate;  is that the restore only recovers the underlying files.  Unless you close and re-open the RocksDB instance, it will retain the current state in memory (it's not exactly clear if the contents are eventually flushed either way, I saw no evidence of it). 

I was also tempted to put a System.exit(0) if an exception occurred after the Restore during the DB close/re-open as the files had been changed so the restarted app would automatically pick them up. Felt a little underhand though. 

Once this is merged, I'll create a release and then roll that change into the existing SC Graph PR.

**Pull request merged**